### PR TITLE
Convert Files from CJS to ESM

### PR DESCRIPTION
**Pull request by mylesw27**
This is working on refactoring files to use ESM for importing and exporting modules. This is one piece of the larger modernization project. This PR intentionally only touches import and exports as to maintain git history and act as a stepping stone in the modernization process. I have started with some of the imports with the instructor pages as those are what I am most familiar with and started updating lib exports as I touched them in the instructor files. Please review thoroughly as many files were touched as part of this PR.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.js
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.js:1]: I can't comment on the line directy, but: the `module.exports` at the end of the file should be rewritten as a default export. Remember that you may have to update places that import this (namely `server.js`). This comment applies to all pages you changed.

**Review by nwalters512 — COMMENTED** [review]
This is looking pretty good, thanks for tackling all this tedious work 🙏 not ready to approve quite yet; the biggest issue is the continued use of `module.exports` in some places.

File: apps/prairielearn/src/lib/aws.js [hunk 1]
```diff
@@ -87,8 +87,8 @@ module.exports.uploadToS3Async = async function (
     logger.verbose(`Uploaded buffer to s3://${s3Bucket}/${s3Path}`);
   }
   return res;
-};
-module.exports.uploadToS3 = util.callbackify(module.exports.uploadToS3Async);
+}
+export const uploadToS3 = util.callbackify(module.exports.uploadToS3Async);
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/lib/aws.js]: I have now removed the use of `module.exports` in several spots. I don't think there should be any more in the files that I have touched.

File: apps/prairielearn/src/lib/file-store.js [hunk 2]
```diff
@@ -3,11 +3,11 @@ const fsPromises = require('fs').promises;
...
 const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
-const sqldb = require('@prairielearn/postgres');
-const { config } = require('./config');
-const sql = sqldb.loadSqlEquiv(__filename);
-const { uploadToS3Async, getFromS3Async } = require('../lib/aws');
+import * as sqldb from '@prairielearn/postgres';
+import { config } from './config';
+import { uploadToS3Async, getFromS3Async } from '../lib/aws';
 
+const sql = sqldb.loadSqlEquiv(__filename);
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/lib/file-store.js:10]: Added spacing to separate the `sql` declaration in commit [77b94ab](https://github.com/PrairieLearn/PrairieLearn/pull/8861/commits/77b94ab3fb8396e9c9627ba521eaa2481987e07b).

File: apps/prairielearn/src/lib/groups.js [hunk 3]
```diff
@@ -349,14 +349,14 @@ module.exports.getGroupRoleReassignmentsAfterLeave = function (groupInfo, leavin
...
 
 /**
  * @param {string} assessmentId
  * @param {string} userId
  * @param {string} authnUserId
  */
-module.exports.leaveGroup = async function (assessmentId, userId, authnUserId) {
+export async function leaveGroup(assessmentId, userId, authnUserId) {
   await sqldb.runInTransactionAsync(async () => {
     const groupId = await module.exports.getGroupId(assessmentId, userId);
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/lib/groups.js]: Removed `module.exports` in commit [77b94ab](https://github.com/PrairieLearn/PrairieLearn/pull/8861/commits/77b94ab3fb8396e9c9627ba521eaa2481987e07b).

File: apps/prairielearn/src/lib/regrading.js [hunk 4]
```diff
@@ -1,226 +1,228 @@
 const ERR = require('async-stacktrace');
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/lib/regrading.js]: We were able to discuss this on Slack but to note a comment here, `require(...)` is still necessary in a few spots as we currently are not able to turn on the `allowSyntheticDefaultImports` flag yet. Once we have converted entirely to ESM, we will be able to import without using `require(...)` for these remaining modules.

File: apps/prairielearn/src/lib/server-jobs-legacy.js [hunk 5]
```diff
@@ -172,7 +172,7 @@ class Job {
 
 let heartbeatIntervalId = null;
 
-module.exports.init = function () {
+export function init() {
   socketServer.io.on('connection', module.exports.connection);
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/lib/server-jobs-legacy.js]: Removed usage of `module.exports` in commit [8492626](https://github.com/PrairieLearn/PrairieLearn/pull/8861/commits/84926260ba52dc1ae4bb9c2508a71afdba8189eb).

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**